filebeat/docs/inputs/input-filestream-file-options.asciidoc

```diff
@@ -615,14 +594,25 @@ file_identity.path: ~
 
 *`inode_marker`*:: If the device id changes from time to time, you must use
 this method to distinguish files. This option is not supported on Windows.
-+
+
 Set the location of the marker file the following way:
 
 [source,yaml]
 ----
 file_identity.inode_marker.path: /logs/.filebeat-marker
 ----
 
+*`fingerprint`*:: To identify files based on their content byte range.
+
+WARNING: In order to use this file identity option, you must enable the <<{beatname_lc}-input-filestream-scan-fingerprint,fingerprint option in the scanner>>. Once this file identity is enabled, changing the fingerprint configuration (offset, length, or other settings) will lead to a global re-ingestion of all files that match the paths configuration of the input.
+
+Please refer to the <<{beatname_lc}-input-filestream-scan-fingerprint,fingerprint configuration for details>>.
+
+[source,yaml]
+----
+file_identity.fingerprint: ~
```

Comment:
Out of curiosity, what is the meaning of the tilde character here?

Reply:
It makes the object `file_identity.fingerprint` to exist. I believe some times when the config is exported/dumped as YAML, it ends up as `file_identity.fingerprint: null`.

We need `file_identity.fingerprint` defined, but it does not have any attribute, nor can it be a primitive type.

Reply:
Aha, thanks. I suspected this was some kind of placeholder, but I was unsure if this was some kind of yaml/parser thing but it seems to be just a project convention, TIL.